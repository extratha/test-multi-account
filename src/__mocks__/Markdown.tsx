const MockMarkdown = (props: any) => {
  return <div {...props} />;
};

jest.mock("react-markdown", () => MockMarkdown);
