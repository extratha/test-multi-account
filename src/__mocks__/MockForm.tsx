import { yupResolver } from "@hookform/resolvers/yup";
import { ReactNode, useEffect } from "react";
import { FormProvider, UseFormReturn, useForm } from "react-hook-form";
import { ObjectSchema } from "yup";

export interface MockFormProps<T = any> {
  initValues: T | undefined;
  resolver?: ObjectSchema<any>;
  children?: ReactNode;
  onInitForm?: (methods: UseFormReturn) => void;
  onSubmit: () => void;
}

export interface MockHandleValueProps {
  name: string;
}

const MockForm = (props: MockFormProps) => {
  const { initValues, resolver, children, onInitForm, onSubmit } = props;
  const methods = useForm({
    reValidateMode: "onSubmit",
    defaultValues: initValues,
    resolver: resolver ? yupResolver(resolver) : undefined,
  });

  useEffect(() => {
    if (onInitForm) onInitForm(methods);
  }, []);

  return (
    <FormProvider {...methods}>
      <form noValidate onSubmit={methods.handleSubmit(onSubmit)}>
        {children}
        <button type="submit" data-testid="submit-button">
          Submit
        </button>
      </form>
    </FormProvider>
  );
};

export default MockForm;
