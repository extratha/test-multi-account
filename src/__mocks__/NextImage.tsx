// eslint-disable-next-line @typescript-eslint/no-unused-vars
const MockImage = ({ src, alt, fill, objectFit, ...props }: any) => {
  const url = src?.src || src;
  // eslint-disable-next-line @next/next/no-img-element
  return <img src={url} alt={alt} style={{ objectFit }} {...props} />;
};

jest.mock("next/image", () => MockImage);
