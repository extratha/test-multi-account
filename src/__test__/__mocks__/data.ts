export const data = {
  test: 'test'
}