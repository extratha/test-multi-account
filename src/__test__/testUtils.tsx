import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
