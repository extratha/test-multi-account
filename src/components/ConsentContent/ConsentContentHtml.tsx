import { styled } from "@mui/material";

export interface ConsentContentHtmlProps {
  name: string;
  html: string;
}

const Html = styled("div")({
  fontSize: "16px",
  lineHeight: "28px",
  "& > h3, & > p": {
    "&:first-of-type": {
      marginTop: "0px",
    },
    "&:last-of-type": {
      marginBottom: "0px",
    },
  },
  "& ul": {
    paddingLeft: "24px",
    listStyleType: "disc",
  },
  "& li": {
    margin: "16px 0px",
  },
  "& ol": {
    listStyleType: "none",
    counterReset: "item",
    padding: "0px",
    "& > li": {
      display: "table-row",
      counterIncrement: "item",
      borderCollapse: "separate",
      borderSpacing: "0px 6px",
      "&:before": {
        display: "table-cell",
        content: `counters(item, ".") ". "`,
        paddingRight: "4px",
        whiteSpace: "nowrap",
      },
    },
    "& > ol, & > ul": {
      paddingLeft: "24px",
      "& > li > ol, & > li > ul": {
        padding: "0px",
      },
    },
  },
  "& table": {
    width: "100%",
    borderCollapse: "collapse",
    tableLayout: "fixed",
    margin: "16px 0px",
    "& li": {
      margin: "0px",
    },
  },
  "& th": {
    height: "48px",
    textAlign: "center",
    color: "#ffffff",
    border: "1px solid #000000",
    borderRightColor: "#ffffff",
    backgroundColor: "#002060",
    "&:last-child": {
      borderColor: "#000000",
    },
  },
  "& td": {
    border: "1px solid #000000",
    padding: "8px",
    verticalAlign: "top",
  },
});

const ConsentContentHtml = ({ name, html }: ConsentContentHtmlProps) => {
  return (
    <Html
      data-testid={`${name}-content`}
      dangerouslySetInnerHTML={{
        __html: html,
      }}
    />
  );
};

export default ConsentContentHtml;