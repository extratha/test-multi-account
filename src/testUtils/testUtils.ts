import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
