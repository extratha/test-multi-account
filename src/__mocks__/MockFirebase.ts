import { mockInputFieldConfig } from "./data";

jest.mock("firebase/app", () => ({
  initializeApp: jest.fn(),
}));

jest.mock("firebase/remote-config", () => ({
  getRemoteConfig: jest.fn().mockReturnValue({ settings: {} }),
  fetchAndActivate: jest.fn().mockResolvedValue(""),
  getValue: jest.fn().mockReturnValue({
    asString: () => mockInputFieldConfig,
  }),
}));
