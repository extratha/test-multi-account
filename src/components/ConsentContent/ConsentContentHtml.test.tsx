import { render } from "@/testUtils/testUtils";

import { mockHtml } from "@/__mocks__/data";
import ConsentContentHtml, { ConsentContentHtmlProps } from "./ConsentContentHtml";

describe("Html", () => {
  let props: ConsentContentHtmlProps;

  beforeEach(() => {
    props = {
      name: "consent",
      html: mockHtml,
    };
  });

  const renderHtml = () => {
    return render(<ConsentContentHtml {...props} />);
  };

  it("should render correctly", () => {
    const { asFragment } = renderHtml();
    expect(asFragment()).toMatchSnapshot();
  });
});
