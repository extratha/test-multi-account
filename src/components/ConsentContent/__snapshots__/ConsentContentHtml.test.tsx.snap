// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 16px;
  line-height: 28px;
}

.emotion-0>h3:first-of-type,
.emotion-0>p:first-of-type {
  margin-top: 0px;
}

.emotion-0>h3:last-of-type,
.emotion-0>p:last-of-type {
  margin-bottom: 0px;
}

.emotion-0 ul {
  padding-left: 24px;
  list-style-type: disc;
}

.emotion-0 li {
  margin: 16px 0px;
}

.emotion-0 ol {
  list-style-type: none;
  counter-reset: item;
  padding: 0px;
}

.emotion-0 ol>li {
  display: table-row;
  counter-increment: item;
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-0 ol>li:before {
  display: table-cell;
  content: counters(item, ".") ". ";
  padding-right: 4px;
  white-space: nowrap;
}

.emotion-0 ol>ol,
.emotion-0 ol>ul {
  padding-left: 24px;
}

.emotion-0 ol>ol>li>ol,
.emotion-0 ol>ul>li>ol,
.emotion-0 ol>ol>li>ul,
.emotion-0 ol>ul>li>ul {
  padding: 0px;
}

.emotion-0 table {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
  margin: 16px 0px;
}

.emotion-0 table li {
  margin: 0px;
}

.emotion-0 th {
  height: 48px;
  text-align: center;
  color: #ffffff;
  border: 1px solid #000000;
  border-right-color: #ffffff;
  background-color: #002060;
}

.emotion-0 th:last-child {
  border-color: #000000;
}

.emotion-0 td {
  border: 1px solid #000000;
  padding: 8px;
  vertical-align: top;
}

<div
    class="emotion-0"
    data-testid="consent-content"
  >
    <p>
      html
    </p>
    

    <table
      style="width:100%"
    >
      
  
      <tbody>
        <tr>
          
    
          <th>
            Company
          </th>
          
    
          <th>
            Contact
          </th>
          
  
        </tr>
        
  
        <tr>
          
    
          <td>
            ASR
          </td>
          
    
          <td>
            Nova
          </td>
          
  
        </tr>
        
  
        <tr>
          
    
          <td>
            SOAP
          </td>
          
    
          <td>
            Star
          </td>
          
  
        </tr>
        

      </tbody>
    </table>
    

    <ul>
      
  
      <li>
        Coffee
      </li>
      

    </ul>
  </div>
</DocumentFragment>
`;
